File: Examples/Python/bluetooth_connection.py
import time

import connection
import helpers
import ximu3

if helpers.yes_or_no("Search for connections?"):
    devices = ximu3.PortScanner.scan_filter(ximu3.PORT_TYPE_BLUETOOTH)

    time.sleep(1)  # wait for OS to release port

    if not devices:
        raise Exception("No Bluetooth connections available")

    print(f"Found {devices[0]}")

    config = devices[0].config

    connection.run(config)
else:
    config = ximu3.BluetoothConnectionConfig("COM1")  # replace with actual connection config

    connection.run(config)

File: Examples/CSharp/x-IMU3-Examples/Examples/BluetoothConnection.cs
namespace Ximu3Examples
{
    class BluetoothConnection : Connection
    {
        public BluetoothConnection()
        {
            if (Helpers.YesOrNo("Search for connections?"))
            {
                var devices = Ximu3.PortScanner.ScanFilter(Ximu3.CApi.XIMU3_PortType.XIMU3_PortTypeBluetooth);

                if (devices.Length == 0)
                {
                    Console.WriteLine("No Bluetooth connections available");
                    return;
                }

                Console.WriteLine("Found " + Ximu3.Helpers.ToString(Ximu3.CApi.XIMU3_device_to_string(devices[0])));

                var config = Ximu3.ConnectionConfig.From(devices[0])!;

                Run(config);
            }
            else
            {
                var config = new Ximu3.BluetoothConnectionConfig("COM1"); // replace with actual connection config

                Run(config);
            }
        }
    }
}
